x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlProcessorAutoscalingDecider.java

```diff
@@ -49,25 +48,35 @@ public MlProcessorAutoscalingCapacity scale(
     ) {
         TrainedModelAssignmentMetadata trainedModelAssignmentMetadata = TrainedModelAssignmentMetadata.fromState(context.state());
 
-        if (hasUnsatisfiedDeployments(trainedModelAssignmentMetadata, mlContext.mlNodes)) {
-            logger.debug(() -> "Computing required capacity as there are partially allocated deployments");
+        final MlProcessorAutoscalingCapacity currentCapacity = computeCurrentCapacity(mlContext.mlNodes, allocatedProcessorsScale);
+        final MlProcessorAutoscalingCapacity requiredCapacity = computeRequiredCapacity(trainedModelAssignmentMetadata).build();
+
+        if (currentCapacity.tierProcessors().roundUp() < requiredCapacity.tierProcessors().roundUp()
+            || currentCapacity.nodeProcessors().roundUp() < requiredCapacity.nodeProcessors().roundUp()) {
+            logger.debug(() -> "required capacity exceeded current capacity, requesting scale up");
             scaleTimer.resetScaleDownCoolDown();
-            return computeRequiredCapacity(trainedModelAssignmentMetadata).setReason(
-                "requesting scale up as there are unsatisfied deployments"
-            ).build();
-        }
 
-        final MlProcessorAutoscalingCapacity currentCapacity = computeCurrentCapacity(mlContext.mlNodes, allocatedProcessorsScale);
```

Comment:
I removed the use of isSatisfied here and replaced it with a direct comparison of the currentCapcity and requiredCapacity as I didn't see any benefit to isSatisfied.